File: template-mirror/atlas-api/src/features/runtime/infrastructure/env-runtime.ts
import {
  createRuntimeContext,
  type RuntimeContext,
} from "../../../core/runtime-context";

/** Infrastructure: load runtime from process env (no domain imports of Bun.env). */
export function loadRuntimeFromEnv(service: string): RuntimeContext {
  return createRuntimeContext({
    service,
    environment: Bun.env.APP_ENV ?? "dev",
    destination: Bun.env.BENCH_DESTINATION ?? "control",
    exporterEndpoint:
      Bun.env.OTEL_EXPORTER_OTLP_ENDPOINT ?? "http://collector:4318",
  });
}

export function listenPort(fallback: number): number {
  return Number(Bun.env.PORT ?? fallback);
}

File: apps/lantern-api/src/features/runtime/infrastructure/env_runtime.py
import os
from src.core.runtime_context import RuntimeContext

def load_runtime(service: str) -> RuntimeContext:
    return RuntimeContext(
        service=service,
        environment=os.environ.get("APP_ENV", "dev"),
        destination=os.environ.get("BENCH_DESTINATION", "control"),
        exporter_endpoint=os.environ.get(
            "OTEL_EXPORTER_OTLP_ENDPOINT", "http://collector:4318"
        ),
    )

def listen_port(fallback: int) -> int:
    return int(os.environ.get("PORT", fallback))
